### PR TITLE
Minor change to switch result checking order so there is no artificial delay.

### PR DESCRIPTION
**Pull request by gjoliver**
## Why are these changes needed?

I noticed this minor issue while debug our regression test.
Existing logic has a 1 sec delay between result fetching and actual result checking, which seems unnecessary.
This PR simply moves some statements around to avoid sleeping in between blocks. Shouldn't have any functionality changes.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [*] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: release/e2e.py [hunk 1]
```diff
@@ -846,17 +846,18 @@ def create_and_wait_for_session(
     start_wait = time.time()
     next_report = start_wait + REPORT_S
     while not completed:
+        session_operation_response = sdk.get_session_operation(
+            sop_id, _request_timeout=30)
+        session_operation = session_operation_response.result
+        completed = session_operation.completed
+
```
Comment (krfricke) [inline release/e2e.py:853]: This move will actually not alter sleeping behavior (as the `not completed` check will only be invoked after the sleep). We should probably just move the `time.sleep(1)` to the top here or just 

```
if not completed:
   time.sleep(1)
```

File: release/e2e.py [hunk 2]
```diff
@@ -893,6 +894,9 @@ def wait_for_session_command_to_complete(create_session_command_result,
     start_wait = time.time()
     next_report = start_wait + REPORT_S
     while not completed:
+        result = sdk.get_session_command(session_command_id=scd_id)
+        completed = result.result.finished_at
+
```
Comment (krfricke) [inline release/e2e.py:899]: Same here, we will still sleep even if the loop is completed here

**Review by krfricke — CHANGES_REQUESTED** [review]
Good catch but I don't think the fix is working, yet, please correct me if I'm wrong

File: release/e2e.py (hunk 2 above)
Comment (gjoliver) [reply release/e2e.py:908]: yep

File: release/e2e.py (hunk 1 above)
Comment (gjoliver) [reply release/e2e.py:861]: oh yeah, good point.
I was too focused on making sure we don't wait a 1 sec before checking the returned result.
this is better.

**Review by gjoliver — COMMENTED** [review]
thanks for the review, totally make sense.

**Comment by gjoliver** [comment]
please take another look

**Review by krfricke — APPROVED** [review]

**Pull request merged**